qubes/events.py

```diff
@@ -157,50 +160,92 @@ def _fire_event_in_order(self, order, event, kwargs):
                     key=(lambda handler: hasattr(handler, 'ha_bound')),
                     reverse=True):
                 effect = func(self, event, **kwargs)
+                # "is True" to rule out unittest.mock.Mock...
```

Comment:
If func is unittest.mock.Mock instance, it do have attribute `ha_async` (in fact: any attribute), which is a new unittest.mock.Mock instance, `bool(unittest.mock.Mock())` is true